lib/moleculer/transporters/redis.rb

```diff
@@ -50,16 +50,40 @@ class Subscriber
         ##
         # Represents a subscription
         class Subscription
+          attr_reader :disconnect_hash, :channel
+          ##
+          # handles processing of messages from redis
+          class MessageProcessor
+            def initialize(subscription)
+              @subscription = subscription
+              @serializer   = Serializers.for(Moleculer.serializer)
+              @packet_type  = Packets.for(@subscription.channel.split(".")[1])
+              @logger       = Moleculer.logger
+            end
+
+            def process(message)
+              return :disconnect if message == @subscription.disconnect_hash
+
+              return nil if message.split(".")[-1] == "disconnect"
+
+              parsed = @serializer.deserialize(message)
+              return nil if parsed["sender"] == Moleculer.node_id
```

Comment:
I think moving the if blocks into methods would help here. That way we aren't worried about the  structure of the message. And all that other stuff we talked about.